src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs/breadcrumb.rs

```diff
@@ -0,0 +1,413 @@
+//! This module provides a clickable view for a single breadcrumb.
+
+use crate::prelude::*;
+
+use super::GLYPH_WIDTH;
+use super::HORIZONTAL_MARGIN;
+use super::VERTICAL_MARGIN;
+use super::TEXT_SIZE;
+
+use enso_frp as frp;
+use ensogl::animation::linear_interpolation;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::shape::text::glyph::system::Line;
+use ensogl::display::shape::text::glyph::system::GlyphSystem;
+use ensogl::display::Sprite;
+use ensogl::gui::component;
+use ensogl::gui::component::Animation;
+use enso_protocol::language_server::MethodPointer;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use std::f32::consts::PI;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const ICON_MARGIN          : f32 = HORIZONTAL_MARGIN;
+const ICON_RADIUS          : f32 = 8.0;
+const ICON_SIZE            : f32 = ICON_RADIUS * 2.0;
+const ICON_RING_WIDTH      : f32 = 2.0;
+const ICON_ARROW_SIZE      : f32 = 5.0;
+const SEPARATOR_SIZE       : f32 = 8.0;
+const SEPARATOR_LINE_WIDTH : f32 = 3.0;
+const SEPARATOR_MARGIN     : f32 = HORIZONTAL_MARGIN;
+const FULL_COLOR           : color::Rgba = color::Rgba::new(1.0, 1.0, 1.0, 0.7);
+const TRANSPARENT_COLOR    : color::Rgba = color::Rgba::new(1.0, 1.0, 1.0, 0.4);
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+mod background {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        () {
+            let bg_color = color::Rgba::new(0.0,0.0,0.0,0.000_001);
+            Plane().fill(bg_color).into()
+        }
+    }
+}
+
+
+
+// ============
+// === Icon ===
+// ============
+
+mod icon {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (opacity:f32) {
+                let outer_circle  = Circle((ICON_RADIUS - 0.5).px());
```

Comment:
this is very bad! It makes the icon not pixel-perfect. This would make the circle aliased on every side. Please do not do it. If you wanted to have your canvas slightly bigger than the icon (for example to make it nice when zooming in) please define PADDING constant and just add it to the canvas and subtract here. But ake it full pixels, not 1/2 of pixel.

Reply:
@wdanilo I did it to fix a weird behavior: if I define a Circle with `ICON_RADIUS` a `ICON_RADIUS x ICON_RADIUS` won't be enough to contain it.

I will follow your suggestion and make the shape bigger.

Reply:
it's OK to make the canvas bigger, but please do not shrink shapes by 0.5 px. This will just make them antialiased. The behavior you describe is not strange. I believe the circle was contained but appeared cropped when zooming in, is that the case? In case you are zooming in, you need bigger canvas for better antialiasing so its normal to create a little bigger canvas, we normally call it PADDING and you will see it everywhere in the codebase of shapes.